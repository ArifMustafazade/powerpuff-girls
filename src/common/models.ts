/** @interfaces */

export interface IApiHandlerResponse<T> {
    status: boolean;
    errorType?: string;
    errorMessage?: string;
    data?: T;
}

interface ILink {
    href: string;
}

interface INetwork {
    id: number;
    name: string;
    country: {
        name: string;
        code: string;
        timezone: string;
    };
}

export interface IEpisode {
    id: number;
    url: string;
    name: string;
    season: number;
    number: number;
    type: string;
    airdate: string;
    airtime: string;
    airstamp: string;
    runtime: number;
    image: {
        medium: string;
        original: string;
    };
    summary: string;
    _links: {
        self: ILink;
    };
}

export interface IShowDetails {
    id: number;
    url: string;
    name: string;
    type: string;
    language: string;
    genres: string[];
    status: string;
    runtime: number;
    premiered: string;
    officialSite: string;
    schedule: {
        time: string;
        days: string[];
    };
    rating: {
        average: number;
    };
    weight: number;
    network: INetwork;
    webChannel: string;
    externals: {
        tvrage: number;
        thetvdb: number;
        imdb: string;
    };
    image: {
        medium: string;
        original: string;
    };
    summary: string;
    updated: number;
    _links: {
        self: ILink;
        previousepisode: ILink;
    };
}
