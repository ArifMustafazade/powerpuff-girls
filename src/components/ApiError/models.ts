/** @interfaces */

export interface IApiErrorProps {
    errorMessage: string;
}
