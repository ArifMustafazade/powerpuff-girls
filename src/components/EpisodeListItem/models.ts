// interface
import { IEpisode } from '../../common/models';

/** @interfaces */

export interface IEpisodeListItemProps {
    episode: IEpisode;
}
