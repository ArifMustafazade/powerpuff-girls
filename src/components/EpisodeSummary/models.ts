// interface
import { IEpisode } from '../../common/models';

/** @interfaces */

export interface IEpisodeSummaryProps {
    episode?: IEpisode;
}
