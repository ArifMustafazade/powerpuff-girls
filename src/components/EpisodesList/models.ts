// interfaces
import { IEpisode } from '../../common/models';

/** @interfaces */
export interface IEpisodesListProps {
    episodes: IEpisode[];
    loading: boolean;
    error: boolean;
    errorMessage: string;
}
