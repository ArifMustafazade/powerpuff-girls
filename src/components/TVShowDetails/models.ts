// interfaces
import { IShowDetails } from '../../common/models';

/** @interfaces */
export interface ITVShowDetailsProps {
    details: IShowDetails;
    error: boolean;
    errorMessage: string;
    loading: boolean;
}
