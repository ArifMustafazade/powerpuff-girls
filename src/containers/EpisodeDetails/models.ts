// interfaces
import { RouteComponentProps } from 'react-router-dom';
import { IEpisodesState } from '../../store/models';
import { IEpisode } from '../../common/models';

/** @interfaces */
export interface IEpisodeDetailsProps extends RouteComponentProps<{ id: string }> {
    episodesList: IEpisodesState;
}

export interface IEpisodeDetailsState {
    episode: IEpisode;
    error: boolean;
    errorMessage: string;
    loading: boolean;
}
