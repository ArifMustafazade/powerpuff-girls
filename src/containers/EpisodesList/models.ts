// interfaces
import { IEpisodesState } from '../../store/models';

/** @interfaces */
export interface IEpisodesListProps {
    episodesList: IEpisodesState;
    onGetEpisodesList: () => void;
}
