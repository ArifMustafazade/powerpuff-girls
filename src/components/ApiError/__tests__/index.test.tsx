import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';

// interfaces
import { IApiErrorProps } from '../models';

// components
import { ApiError } from '../';

/** tests */
describe('<ApiError />', () => {
    let wrapper: ShallowWrapper<IApiErrorProps>;

    const errorMessage = 'TEST_ERROR';

    beforeEach(() => {
        wrapper = shallow(<ApiError errorMessage={errorMessage} />);
    });

    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should render with correct error message', () => {
        expect(wrapper.find('.error__message').text()).toBe(errorMessage);
    });
});
