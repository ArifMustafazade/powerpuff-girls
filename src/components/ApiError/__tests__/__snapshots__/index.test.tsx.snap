// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApiError /> should match snapshot 1`] = `
<div
  className="error"
>
  <p
    className="error__message"
  >
    TEST_ERROR
  </p>
</div>
`;
